src/components/dataset/query/sidebar/AppliedFilterList.jsx

```diff
@@ -0,0 +1,166 @@
+import React from 'react';
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { withStyles } from '@material-ui/core/styles';
+
+import Card from '@material-ui/core/Card';
+import List from '@material-ui/core/List';
+import ListSubheader from '@material-ui/core/ListSubheader';
+import ListItem from '@material-ui/core/ListItem';
+import { applyFilters } from '../../../../actions';
+import { Chip, Collapse, Badge } from '@material-ui/core';
+import { ExpandLess } from '@material-ui/icons';
+
+const styles = theme => ({
+  filterHeader: {
+    paddingTop: theme.spacing(1),
+    paddingBottom: theme.spacing(0.5),
+    lineHeight: 'inherit',
+  },
+  rangeInfo: {
+    display: 'inline-block',
+  },
+  inline: {
+    margin: '2px',
+  },
+  tableName: {
+    justifyContent: 'space-between',
+    fontWeight: 500,
+  },
+  box: {
+    margin: `${theme.spacing(1)}px ${theme.spacing(2)}px ${theme.spacing(1)}px ${theme.spacing(
+      2,
+    )}px`,
+  },
+});
+
+const StyledBadge = withStyles(theme => ({
+  badge: {
+    right: 12,
+    top: 0,
+    backgroundColor: theme.palette.primary.lightContrast,
+  },
+}))(Badge);
+
+export class AppliedFilterList extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: true,
+    };
+  }
+
+  static propTypes = {
+    classes: PropTypes.object,
+    filterData: PropTypes.object,
+    dataset: PropTypes.object,
+    dispatch: PropTypes.func.isRequired,
+    selected: PropTypes.string,
+    table: PropTypes.string,
+  };
+
+  handleExpand = () => {
+    const { open } = this.state;
+    this.setState({ open: !open });
+  };
+
+  clearFilter = (table, filter, datum) => {
+    const { dispatch, filterData, dataset, selected } = this.props;
+    const { relationships } = dataset.schema;
+    const clonedData = _.cloneDeep(filterData);
+    const clonedFilter = clonedData[table][filter];
+    const filterValue = clonedFilter.value;
+
+    if (clonedFilter.type !== 'range') {
+      if (Array.isArray(filterValue)) {
+        filterValue.splice(filterValue.indexOf(datum), 1);
+      }
+      if (_.isPlainObject(filterValue)) {
+        delete filterValue[datum];
+      }
+    }
+
+    if (_.isEmpty(filterValue) || clonedFilter.type === 'range') {
```

Comment:
If `clonedFilter.type === 'range'`, what is `filterValue`?

Reply:
It's an array, with `filterValue[0]` being the lower bound of the range and `filterValue[1]` equal to the upper bound. I think it comes from `handleChange` on our range slider component. In this case we don't want to be able to delete just one item from the array, so we remove the entire filter. Hope that makes sense!